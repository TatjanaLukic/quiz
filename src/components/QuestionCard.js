import React from "react";

const styleQuestion = {
  color: "white"
};

const styleAnswers = {
  color: "white",
  margin: "5px"
};

const QuestionCard = props => {
  const onSelectHandler = event => {
    props.userAnswersUpdate(event.target.name, event.target.value);
  };

  return (
    <div className="grouped fields" style={{ marginTop: "5px" }}>
      <h3 style={styleQuestion}>{props.question.q}</h3>
      <div className="field">
        <div className="ui radio checkbox">
          <input
            type="radio"
            name={props.question.id}
            value="A"
            onChange={onSelectHandler}
          />
          <label style={styleAnswers}>{props.question.A}</label>
        </div>
      </div>
      <div className="field">
        <div className="ui radio checkbox">
          <input
            type="radio"
            name={props.question.id}
            value="B"
            onChange={onSelectHandler}
          />
          <label style={styleAnswers}>{props.question.B}</label>
        </div>
      </div>
      <div className="ui divider"></div>
    </div>
  );
};

export default QuestionCard;
