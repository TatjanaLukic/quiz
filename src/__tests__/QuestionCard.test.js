import React from "react";

import { configure, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";

import QuestionCard from "../components/QuestionCard";

configure({ adapter: new Adapter() });

const question = {
  id: 1,
  q: "Mock question?",
  A: "Mock answer 1.",
  B: "Mock answer 2.",
  answer: "A",
  category: "basic",
};
const {id} = question;
const userAnswersUpdate = () =>{};

describe("<QuestionCard/>", () => {
  it("spanshot matches", () => {

    const tree = shallow(<QuestionCard id question userAnswersUpdate/>);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
