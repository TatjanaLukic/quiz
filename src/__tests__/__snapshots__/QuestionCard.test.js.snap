// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuestionCard/> spanshot matches 1`] = `
<div
  className="grouped fields"
  style={
    Object {
      "marginTop": "5px",
    }
  }
>
  <h3
    style={
      Object {
        "color": "white",
      }
    }
  />
  <div
    className="field"
  >
    <div
      className="ui radio checkbox"
    >
      <input
        onChange={[Function]}
        type="radio"
        value="A"
      />
      <label
        style={
          Object {
            "color": "white",
            "margin": "5px",
          }
        }
      />
    </div>
  </div>
  <div
    className="field"
  >
    <div
      className="ui radio checkbox"
    >
      <input
        onChange={[Function]}
        type="radio"
        value="B"
      />
      <label
        style={
          Object {
            "color": "white",
            "margin": "5px",
          }
        }
      />
    </div>
  </div>
  <div
    className="ui divider"
  />
</div>
`;
